### ISSUE TITLE
How to change audio codec options? 

### ISSUE THREAD

**Issue by happysmash27** [ammen99/wf-recorder#100]
I would like to record flac in 48000HZ. Is it currently possible to do this, and if not, would it be feasible for me, as a person with only basic C++ knowledge (but knowledge of C and the FFMPEG command line), to implement this before tomorrow?

**Comment by happysmash27** [comment]
I'm surprised! I'm guessing you learned ffmpeg mostly as an API instead of a program? I use ffmpeg in the command line for many things (even some video editing occasionally), so the options I added that are not based on the current options are mostly based on ffmpeg ones, or are based on both. 

This is the command I used to use to record my video output: 
```
ffmpeg -f alsa -ac 2 -ar 44100 -i loopout -f alsa -i pulse -filter_complex amix=inputs=2 -f x11grab -s 1280x1024 -i :0.0 -r 60 -acodec pcm_s16le -vcodec libx264 -crf 0 -preset ultrafast output.mkv
ffmpeg -i output.mkv -vcodec libx264 final.avi
rm output.mkv
```
Looking back on it, I probably would have used different options today, but that is what I used back then. 

If it wasn't a conscious decision to use --codec instead of --vcodec, I would recommend the latter, perhaps preserving --codec as an alias to it for backwards compatibility, just to be a bit more consistent with ffmpeg and with having a codec option for both audio and video. Perhaps that should be a separate issue/pull request, though.

**Comment by ammen99** [comment]
I'd rather not change what already exists, but maybe we can add vcodec for compatibility (in a separate PR). And yes, I learned mostly as an API, and I have almost zero experience with the actual ffmpeg command :)

**Comment by happysmash27** [comment]
It took me a bit too long to implement the change of sample rate, but in testing, apparently it's actually _needed_ even with all the default formats. All my audio is out of sync, since it is 48000HZ but wf-recorder is interpreting it as 44100HZ. I already recorded the thing, and guess I will have to fix it in post. Hopefully I can fix the issue today after that.

**Comment by happysmash27** [comment]
On second thought, after getting the sample format to work and more testing, I think that is an unrelated issue.

**Comment by happysmash27** [comment]
I have made a pull request, but still have a function I am still trying to figure out in order to make it integrate best with the rest of the code. May I ask what the purpose of this function is: 

```
static enum AVSampleFormat get_codec_sample_fmt(AVCodec *codec)
{
    int i = 0;
    if (!codec->sample_fmts)
        return AV_SAMPLE_FMT_S16;
    while (1) {
        if (codec->sample_fmts[i] == -1)
            break;
        if (av_get_bytes_per_sample(codec->sample_fmts[i]) >= 2)
            return codec->sample_fmts[i];
        i++;
    }
    return codec->sample_fmts[0];
}
```
?

**Comment by truesilver92** [comment]
@happysmash27 I think the best way to add this functionallity, instead of adding another interface that essentially exposes slightly more of ffmpeg's interface, just change the scope of the project to just adding an input device to ffmpeg https://github.com/ammen99/wf-recorder/issues/139

**Comment by ammen99** [comment]
Fixed in #177
